pkg/operator/starter.go

```diff
@@ -97,6 +98,27 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 		return err
 	}
 
+	desiredVersion := status.VersionForOperatorFromEnv()
+	missingVersion := "0.0.1-snapshot"
+
+	// By default, this will exit(0) the process if the featuregates ever change to a different set of values.
```

Comment:
wow, may I know why the process will exit if the featuregates change?

Reply:
> wow, may I know why the process will exit if the featuregates change?

Simplest way to be sure that every control loop is reconsidered.  And the operators are restartable at any time. And featuregates can only change once in the lifetime of the cluster.